app/controllers/api/v1/shared_accesses_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class SharedAccessesController < ApplicationController
+      skip_before_action :verify_authenticity_token
+      before_action :find_room
+
+      # POST /api/v1/shared_accesses/room/friendly_id.json
+      def create
+        users = User.where(id: params[:users])
+
+        users.each do |user|
+          SharedAccess.find_or_create_by!(user_id: user.id, room_id: @room.id) if user.room_shareable?(@room)
+        end
```

Comment:
Performance wise - this isn't very good. What I would do instead is filter the users first, and the create them all in one shot instead of looping through them manually

Reply:
And why the `find_or_create_by`?